docs/src/recipes/cycle-detection.asciidoc

```diff
@@ -48,6 +48,25 @@ the length of the cycle is known to be three and there is no need to exceed that
 cycle. It returned three, because there was one for each vertex that started the cycle (i.e. one for `A`, one for `B`
 and one for `C`). This next line introduce deduplication to only return unique cycles.
 
+Note that these traversals won't detect self-loops (vertices directly connected to themselves).
+To do so, you would need to `.emit()` a Traverser before the repeat()-loop.
+
+[gremlin-groovy]
+----
+g.addV().property(id,'a').as('a').
+  addV().property(id,'b').as('b').
+  addV().property(id,'c').as('c').
+  addV().property(id,'d').as('d').
+  addE('knows').from('a').to('b').
+  addE('knows').from('b').to('c').
+  addE('knows').from('c').to('a').
+  addE('knows').from('a').to('d').
+  addE('knows').from('c').to('d').
+  addE('self').from('a').to('a').iterate()
+g.V().as('a').emit().repeat(out().simplePath()).times(2).
+  where(out().as('a')).path()
```

Comment:
I would tweaks the traversal a little bit to make the output easier to grasp:
```
gremlin> g.V().as('a').
......1>   emit().
......2>     repeat(outE().inV().simplePath()).
......3>     times(2).
......4>   outE().inV().where(eq('a')).
......5>   path().
......6>     by(id).
......7>     by(label)
==>[a,self,a]
==>[a,knows,b,knows,c,knows,a]
==>[b,knows,c,knows,a,knows,b]
==>[c,knows,a,knows,b,knows,c]
```

Reply:
Good point!

Reply:
@dkuppitz what is the added value of `outE().inV()` compared to `out()`?

Reply:
Really just readability. To me `[a,self,a]` looks better than just `[a,a]`.